mailpoet/lib/Newsletter/ViewInBrowser/ViewInBrowserController.php

```diff
@@ -103,20 +101,4 @@ private function getSubscriber(array $data): SubscriberEntity {
 
     return $subscriber ?? new SubscriberEntity();
   }
-
-  private function getQueue(NewsletterEntity $newsletter, array $data): ?SendingQueueEntity {
-    // queue is optional; try to find it if it's not defined and this is not a welcome email
-    if ($newsletter->getType() === NewsletterEntity::TYPE_WELCOME) {
-      return null;
-    }
-
-    // reset queue when automatic email is being previewed
-    if ($newsletter->getType() === NewsletterEntity::TYPE_AUTOMATIC && !empty($data['preview'])) {
-      return null;
-    }
-
-    return !empty($data['queue_id'])
```

Comment:
Reading the Jira ticket and the commit messages, it is clear to me why it was necessary to change this part of the function and return `null` if `queue_id` is not set. But it is not clear to me why the second commit is necessary and why we don't need any more special handling for welcome and automatic emails. Could you please elaborate?

Reply:
@rodrigoprimo The second commit is not necessary, it's more like a cleanup. I think those IF's weren't necessary in the first place. They are just hotfixes on the original code, which used to be like this:

```php
if($data->newsletter->type !== Newsletter::TYPE_WELCOME) {
  $queue = ...; // fetch by queue ID or fall back to any queue
}

// now here we already have a fallback queue so we have to remove it
if($data->newsletter->type === Newsletter::TYPE_AUTOMATIC && !empty($data->preview)) {
  $data->queue = false;
}
```

Then, it was refactored to a form where the `fetch by queue ID or fall back to any queue` is actually **after** the IFs, and when we remove that fallback, those IFs make no more sense.

The second commit has in the message links to the original changes:

> Originally, this was done in an IF: https://github.com/mailpoet/mailpoet/commit/32f7d7771f2dcbd6e24a1dcc7912d7aacbd124c3
> Then another hotfix added here: https://github.com/mailpoet/mailpoet/commit/116ee51497ae33456dbb5f804bee0f0754587d61

Reply:
Thanks for the additional details. I agree that it seems those ifs we not necessary in the first place.